### PR TITLE
feat: add method for decoding sealed results

### PR DESCRIPTION
**Pull request by TheUnderScorer**

### CODE REVIEW

File: sdk/sealed/unseal_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	sealedResult := base64Decode("noXc7SXO+mqeAGrvBMgObi/S0fXTpP3zupk8qFqsO/1zdtWCD169iLA3VkkZh9ICHpZ0oWRzqG0M9/TnCeKFohgBLqDp6O0zEfXOv6i5q++aucItznQdLwrKLP+O0blfb4dWVI8/aSbd4ELAZuJJxj9bCoVZ1vk+ShbUXCRZTD30OIEAr3eiG9aw00y1UZIqMgX6CkFlU9L9OnKLsNsyomPIaRHTmgVTI5kNhrnVNyNsnzt9rY7fUD52DQxJILVPrUJ1Q+qW7VyNslzGYBPG0DyYlKbRAomKJDQIkdj/Uwa6bhSTq4XYNVvbk5AJ/dGwvsVdOnkMT2Ipd67KwbKfw5bqQj/cw6bj8Cp2FD4Dy4Ud4daBpPRsCyxBM2jOjVz1B/lAyrOp8BweXOXYugwdPyEn38MBZ5oL4D38jIwR/QiVnMHpERh93jtgwh9Abza6i4/zZaDAbPhtZLXSM5ztdctv8bAb63CppLU541Kf4OaLO3QLvfLRXK2n8bwEwzVAqQ22dyzt6/vPiRbZ5akh8JB6QFXG0QJF9DejsIspKF3JvOKjG2edmC9o+GfL3hwDBiihYXCGY9lElZICAdt+7rZm5UxMx7STrVKy81xcvfaIp1BwGh/HyMsJnkE8IczzRFpLlHGYuNDxdLoBjiifrmHvOCUDcV8UvhSV+UAZtAVejdNGo5G/bz0NF21HUO4pVRPu6RqZIs/aX4hlm6iO/0Ru00ct8pfadUIgRcephTuFC2fHyZxNBC6NApRtLSNLfzYTTo/uSjgcu6rLWiNo5G7yfrM45RXjalFEFzk75Z/fu9lCJJa5uLFgDNKlU+IaFjArfXJCll3apbZp4/LNKiU35ZlB7ZmjDTrji1wLep8iRVVEGht/DW00MTok7Zn7Fv+MlxgWmbZB3BuezwTmXb/fNw==")
+	key := base64Decode("p2PA7MGy5tx56cnyJaFZMr96BCFwZeHjZV2EqMzTq53=")
+
+	keys := []DecryptionKey{
+		{
+			Key:       base64Decode("p2PA7MGy5tx56cnyJaFZMr96BCFwZeHjZV2EqMvTq54="),
+			Algorithm: AlgorithmAES256GCM,
+		},
+		{
```
Comment (ilfa) [inline sdk/sealed/unseal_test.go:62]: Please add one more bad key with a wrong size

File: template/README.mustache [hunk 2]
```diff
@@ -128,6 +128,11 @@ func main() {
 }
 ```
 
+## Sealed results
+
+This SDK provides utility methods for decoding sealed results.
+To learn more, refer to example located in [example/sealedResults.go](./example/sealedResults.go).
```
Comment (ilfa) [inline template/README.mustache:134]: The example is not really big, let's add the minimum variant to the readme.

**Review by ilfa — CHANGES_REQUESTED** [review]
We need to test more bad scenarios, you can just check code coverage for this file.
image

**Comment by ilfa** [comment]
I forgot to mention.
We need documentation for the API too.

File: sdk/sealed/unseal_test.go (hunk 1 above)
Comment (TheUnderScorer) [reply sdk/sealed/unseal_test.go:62]: Done!

File: example/sealedResults.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			Algorithm: sealed.AlgorithmAES256GCM,
+		},
+	}
+	unsealedResponse, err := sealed.UnsealEventsResponse(sealedResult, keys)
+	if err != nil {
+		fmt.Println("Unseal error:", err)
+		return
+	}
+
+	responseJsonData, err := json.MarshalIndent(unsealedResponse, "", "  ")
```
[the remark this answers is not in the document]
Comment (TheUnderScorer) [reply example/sealedResults.go]: Done! I've extracted an utility function for printing event response using known structure

**Comment by ilfa** [comment]
@TheUnderScorer I managed to broke it (panic) with next negative test:

```go
	t.Run("with empty data", func(t *testing.T) {
		sealedResult := []byte("")
		key := base64Decode("p2PA7MGy5tx56cnyJaFZMr96BCFwZeHjZV2EqMvTq53=")

		keys := []DecryptionKey{
			{
				Key:       key,
				Algorithm: AlgorithmAES256GCM,
			},
		}
		_, err := UnsealEventsResponse(sealedResult, keys)

		assert.Error(t, err, ErrInvalidHeader)
	})
```

**Comment by ilfa** [comment]
@TheUnderScorer I found one more missing testcase:

```go
	t.Run("with bad nonce", func(t *testing.T) {
		sealedResult := []byte{0x9E, 0x85, 0xDC, 0xED, 0xAA, 0xBB, 0xCC}
		key := base64Decode("p2PA7MGy5tx56cnyJaFZMr96BCFwZeHjZV2EqMvTq53=")

		keys := []DecryptionKey{
			{
				Key:       key,
				Algorithm: AlgorithmAES256GCM,
			},
		}
		_, err := UnsealEventsResponse(sealedResult, keys)

		assert.Error(t, err)
		assert.IsType(t, err, &AggregatedUnsealError{})

		var aggregateError *AggregatedUnsealError
		errors.As(err, &aggregateError)

		assert.Len(t, aggregateError.UnsealErrors, 1)

		unsealError := aggregateError.UnsealErrors[0]
		assert.ErrorContains(t, unsealError.Error, "nonce")
	})
```

**Review by ilfa — APPROVED** [review]

**Pull request merged**